### PR TITLE
Make enable-sp=nonblock behavior consistent

### PR DESCRIPTION
**Pull request by kaleb-himes**
# Description
Enforces consistent behavior with attempting to configure either `--enable-sp=nonblock --enable-sp-asm` vs `--enable-sp=nonblock,<any other option at all whatsoever> --enable-sp-asm`. Previously the first would pass and the second would report a configuration error, now they both behave the same.

# Testing

https://cloud.wolfssl-test.com/jenkins/view/master-PRBs/job/PRB-sp-options-fasanitize-A-v2/6/ - The case where one passed

https://cloud.wolfssl-test.com/jenkins/view/master-PRBs/job/PRB-sp-options-fasanitize-A-v2/7/ - The case where both report the same configuration error

# Checklist

 - [X] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Review by dgarske — APPROVED** [review]

**Comment by kaleb-himes** [comment]
NOTE: relates to https://github.com/wolfSSL/testing/pull/368

File: configure.ac [hunk 1]
```diff
@@ -6790,6 +6790,7 @@ if test "$ENABLED_SP_NO_MALLOC" = "yes"; then
     AM_CCASFLAGS="$AM_CCASFLAGS -DWOLFSSL_SP_NO_MALLOC"
 fi
 if test "$ENABLED_SP_NONBLOCK" = "yes"; then
+    ENABLED_SP=yes
```
Comment (SparkiDev) [inline configure.ac:6793]: Does nothing.
What will work is adding a check for ENABLES_SP being just "nonblack" and add ",yes" to it, before the parsing ENABLED_SP.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (kaleb-himes) [reply configure.ac:6793]: Thank you so much! Will update as soon as possible.

**Comment by kaleb-himes** [comment]
Using this PR as a guinea pig for the 8 new tests now that they are in production: retest this please

**Pull request closed without merging**